src/python/pants/rules/core/fmt.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+
+from pants.base.build_environment import get_buildroot
+from pants.engine.console import Console
+from pants.engine.fs import Digest, FilesContent
+from pants.engine.goal import Goal
+from pants.engine.legacy.graph import HydratedTargets
+from pants.engine.legacy.structs import (
+  PythonAppAdaptor,
+  PythonBinaryAdaptor,
+  PythonTargetAdaptor,
+  PythonTestsAdaptor,
+)
+from pants.engine.rules import console_rule, union
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class FmtResult(datatype([
+  ('digest', Digest),
+  ('stdout', str),
+  ('stderr', str),
+])):
```

Comment:
Conver to dataclass

```python
from dataclasses import dataclass

@dataclass(frozen=True)
class FmtResult:
  digest: Digest
  stdout: str
  stderr: str
```

Make sure you include `3rdparty/python:dataclasses` in BUILD.